File: 2734. Lexicographically Smallest String After Substring Operation/Solution.cpp
class Solution {
public:
    string smallestString(string s) {
        bool change = false;
        int l = s.length(), i =  0;
        while (i < l && s[i] == 'a') i += 1;
        if (i == l) return s.substr(0, l - 1) + "z";
        while (i < l && s[i] != 'a') {
            s[i] = (char) (s[i] - 1);
            i += 1;
        }
        return s;
        
    }
};

File: 2734. Lexicographically Smallest String After Substring Operation/Solution.py
class Solution:
    def smallestString(self, s: str) -> str:
        change, l, i = False, len(s), 0
        ch = [c for c in s]
        while i < l and ch[i] == 'a': i += 1
        if i == l: return s[:-1] + 'z'
        while i < l and ch[i] != 'a':
            ch[i] = chr(ord(ch[i]) - 1)
            i += 1
        return ''.join(ch)
